src/Core/Component.js

```diff
@@ -343,31 +444,34 @@ function getComponentContext (element, props, state, context) {
  * @param {Object} props
  * @param {Object} state
  * @param {Object} context
- * @return {Object?}
+ * @return {boolean?}
  */
-function getLifecycleData (element, name, props, state, context) {
+function getLifecycleReceive (element, name, props, state, context) {
 	try {
-		return element.owner[name].call(element.instance, props, state, context)
+		getLifecycleState((element.work = SharedWorkIntermediate, element), element.owner[name](props, context))
```

Comment:
I'm going to stop commenting about this style of syntax, but note that I think it's significantly detrimental to comprehension to nest an unused variable assignment into an expression like this